Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -94,19 +95,48 @@ public function __construct(
      */
     public function handle(CommandInterface $command): void
     {
-        // the commands only calculate which events they want to have published, but do not do the
-        // publishing themselves
-        $eventsToPublishOrGenerator = $this->commandBus->handle($command);
-
-        if ($eventsToPublishOrGenerator instanceof EventsToPublish) {
-            $eventsToPublish = $this->enrichEventsToPublishWithMetadata($eventsToPublishOrGenerator);
-            $this->eventPersister->publishEvents($this, $eventsToPublish);
-        } else {
-            foreach ($eventsToPublishOrGenerator as $eventsToPublish) {
-                assert($eventsToPublish instanceof EventsToPublish); // just for the ide
-                $eventsToPublish = $this->enrichEventsToPublishWithMetadata($eventsToPublish);
-                $this->eventPersister->publishEvents($this, $eventsToPublish);
+        $toPublish = $this->commandBus->handle($command);
+
+        if ($toPublish instanceof EventsToPublish) {
+            // simple case
+            $eventsToPublish = $this->enrichEventsToPublishWithMetadata($toPublish);
+            if ($eventsToPublish->events->isEmpty()) {
+                return;
+            }
```

Comment:
This check could be moved before the enrichment I suppose
 But it's not important and it will hopefully be removed with my other pr anyways

Reply:
adjusted :)